dog.md

```diff
@@ -0,0 +1,208 @@
+```
+
+File-able Fields
+----------------
+
+These parameters are controlled by governance:
+
+-   `vow`:  Debt Engine
+-   `hole`: Max DAI needed to cover debt+fees of active auctions
+-   `chop`: Liquidation Penalty
+-   `clip`: Liquidator
+
+```k
+    syntax DogAuthStep ::= "file" DogFile
+ // -------------------------------------
+
+    syntax DogFile ::= "vow"  Address
+                     | "Hole" Rad
+                     | "hole" String Rad
+                     | "chop" String Wad
+                     | "clip" String Address
+ // -----------------------------------------
+    rule <k> Dog . file vow DOG_VOW => . ... </k>
+         <dog-vow> _ => DOG_VOW </dog-vow>
+
+    rule <k> Dog . file Hole HOLE => . ... </k>
+         <dog-hole> _ => HOLE </dog-hole>
+
+    rule <k> Dog . file hole ILK HOLE => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... hole: (_ => HOLE) ) ... </dog-ilks>
+
+    rule <k> Dog . file chop ILK CHOP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... chop: (_ => CHOP) ) ... </dog-ilks>
+      requires CHOP >=Wad wad(1)
+
+    rule <k> Dog . file clip ILK CLIP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: (_ => CLIP) ) ... </dog-ilks>
+         <clip-ilk> CLIP_ILK </clip-ilk>
+      requires ILK ==K CLIP_ILK
+```
+
+Dog Semantics
+-------------
+
+```k
+    syntax DogStep ::= "bark" String Address Address
+ // ------------------------------------------------
+    rule <k> Dog . bark ILK URN KPR
+    =>  #let ROOM = minRad( ( DOG_HOLE -Rad DOG_DIRT), (HOLE -Rad  DIRT) ) #in (
+        #let DART_INITIAL = minWad( ART, ( ( ROOM /Rate RATE) /Wad CHOP) ) #in (
+        #let DART_FINAL = (#if (ART >Wad DART_INITIAL) andBool ( ((ART -Wad DART_INITIAL) *Rate RATE) <Rad DUST ) #then ART #else   DART_INITIAL #fi) #in (
+        #let NON_DUSTY = (#if (ART >Wad DART_INITIAL) andBool ( ((ART -Wad DART_INITIAL) *Rate RATE) <Rad DUST ) #then true #else (DART_INITIAL *Rate RATE) >=Rad DUST  #fi) #in (
+        #let DINK = (INK *Wad DART_FINAL) /Wad ART #in (
+        #let DUE = DART_FINAL *Rate RATE #in (
+        #let TAB = DUE *RadWad2Rad CHOP #in (
+    NON_DUSTY
+    ~> KPR
+    ~> TAB
+    ~> emitBark ILK URN DINK DART_FINAL DUE CLIP 0 ) ) ) ) ) ) )
+    ... </k>
+         <dog-live> DOG_LIVE  </dog-live>
+         <dog-dirt> DOG_DIRT  </dog-dirt>
+         <dog-hole> DOG_HOLE  </dog-hole>
+         <vat-urns> ... { ILK, URN } |-> Urn( INK, ART ) ...                                                </vat-urns>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: CLIP:ClipContract, chop: CHOP, hole: HOLE,  dirt: DIRT ) ... </dog-ilks>
+         <vat-ilks> ... ILK |-> Ilk( ... rate: RATE, spot: SPOT, dust: DUST ) ...                           </vat-ilks>
+      requires DOG_LIVE ==Bool true
+      andBool ( ( SPOT >Ray ray(0) )
+      andBool ( (INK *Rate SPOT ) <Rad (ART *Rate RATE) ) )
+      andBool ( (DOG_HOLE >Rad DOG_DIRT )
+      andBool ( HOLE >Rad DIRT ) )
```

Comment:
I think we need `NON_DUSTY ==Bool true` as a side condition here as well.

Reply:
I think I captured all those `requires` clauses

Reply:
I'm assuming the below is some sort of continuation of processing the `bark` logic so there's need to require `NON_DUSTY` here. If that's at least roughly correct go ahead and resolve.

Reply:
Yep yep, further explained in the big answer of the `emitBark` event